framgia/sph-els-joshua
# [FE/BE] Implement User Register/Login Integration

### **Asana Link** 

- https://app.asana.com/0/1202627219907022/1202641655555610/f

### **Definition of Done**

- [x] [BE] Install Laravel Breeze Starter Kits
- [x] [FE] Install axios, swr and react-toastify deps
- [x] [FE] Implement auth hook for login and register integration
- [x] [FE] Implement auth-protected file to check if user is authenticated or !
- [x] [FE] Display auth validation errors in login/register
 
### **Notes**

- I modify `breeze-next` auth hooks with typescript and some other property prop `types` are putted into `any`
Ref: https://github.com/laravel/breeze-next/blob/master/src/hooks/auth.js
- Currently, `breeze-next` recommend using localhost during local development of backend and frontend to avoid CORS "Same-Origin" issues.
Ref: https://github.com/laravel/breeze-next

### **Scenarios/Test Cases**

go to `backend` file
- create `.env` from your root file
- Make sure you start your xampp or any provider you have
- Set up your `DB_DATABASE=your-database-name`

cd backend
- composer install
- php artisan migrate
- php artisan serve

go to `frontend` file
- create `.env.local` from your root file
- paste it `NEXT_PUBLIC_BACKEND_URL=http://localhost:8000/`

cd frontend
- yarn install || npm install
- yarn dev || npm run dev

### **Screenshots**
![image](https://user-images.githubusercontent.com/38458781/180598473-a4d229ef-ae71-493d-b0cc-53628a77b0df.png)
![image](https://user-images.githubusercontent.com/38458781/180598490-ce5098fa-f4a5-4003-9bfa-7a841b60c3f0.png)
